### PR TITLE
Add reference for Copyable

### PR DESCRIPTION
**Pull request by amartini51**
Fixes: rdar://126377559

### CODE REVIEW

**Comment by amartini51** [comment]
Unlike the reference for [Sendable](https://developer.apple.com/documentation/swift/sendable) I didn't add a cross reference to "The Swift Programming Language" at the end of the overview.  I'm not sure if there will just one place in TSPL that discusses noncopyable types, or if the content will end up needing to be spread across multiple chapters.  If we do find a good place to link folks to, to dive into more detail, we can come back and add on later one.

File: stdlib/public/core/Misc.swift [hunk 1]
```diff
@@ -169,6 +169,65 @@ func _rethrowsViaClosure(_ fn: () throws -> ()) rethrows {
...
 }
 
+/// A type whose values can be implicitly or explicitly copied.
+///
+/// Conforming to this protocol indicates that a type's value can be copied;
+/// this protocol doesn’t have any required methods or properties.
+/// You don't generally need to write an explicit conformance to `Copyable`.
+/// The following places implicitly include `Copyable` conformance:
+///
+/// * Structures declarations,
```
Comment (slavapestov) [inline stdlib/public/core/Misc.swift:179]: should this read "structure declarations"?

Comment (amartini51) [reply stdlib/public/core/Misc.swift]: Yes, thanks — fixed in 587c79b6f1f.

**Comment by amartini51** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Misc.swift [hunk 2]
```diff
@@ -169,6 +169,65 @@ func _rethrowsViaClosure(_ fn: () throws -> ()) rethrows {
...
+///
+/// * Structure declarations,
+///   unless it has a noncopyable stored property
+/// * Enumeration declarations,
+///   unless it has a case whose associated value isn't copyable
+/// * Class declarations
+/// * Actor declarations
+/// * Protocol declarations
+/// * Associated type declarations
+/// * The `Self` type in a protocol extension
```
Comment (kavon) [inline stdlib/public/core/Misc.swift:187]: Did we consider whether to call out the generic parameters in an extension of a generic type? Something like:

```suggestion
/// * The `Self` type in a protocol extension
/// * The generic parameters of a nominal type in an extension
```

**Review by kavon — APPROVED** [review]
Overall I think this is good! Only one minor suggestion.

File: stdlib/public/core/Misc.swift (hunk 2 above)
Comment (slavapestov) [reply stdlib/public/core/Misc.swift:187]: s/nominal type/extended type/

Comment (amartini51) [reply stdlib/public/core/Misc.swift:187]: Changed in 6fa34bdabf7

**Comment by amartini51** [comment]
@swift-ci Please smoke test

**Review by bjlanier — APPROVED** [review]

**Comment by amartini51** [comment]
@swift-ci Please smoke test.

**Review by glessard — APPROVED** [review]

**Comment by amartini51** [comment]
@swift-ci Please smoke test.

**Comment by amartini51** [comment]
@swift-ci Please smoke test.

**Pull request merged**